src/lib/io/sol-iio.c

```diff
@@ -870,6 +889,8 @@ sol_iio_close(struct sol_iio_device *device)
     SOL_NULL_CHECK(device);
 
     SOL_PTR_VECTOR_FOREACH_REVERSE_IDX (&device->channels, channel, i) {
+        /* Deactivate device firmware/hardware after use */
+        set_channel_enabled(device, channel->name, false);
```

Comment:
Sounds fair: it's better to centralise this operation. So if code flow changes in future, any call to `iio_del_channel` is sure to do a proper clean up.